File: source/backend/src/main/java/ch/unisg/backend/core/application/NonFunctionalRequirementService.java
package ch.unisg.backend.core.application;

import ch.unisg.backend.core.domain.entities.classes.ar.NonFunctionalRequirement;
import ch.unisg.backend.core.port.in.NonFunctionalRequirementUseCase;
import ch.unisg.backend.core.port.in.command.classes.NonFunctionalRequirementCommand;
import ch.unisg.backend.core.port.in.query.NonFunctionalRequirementQuery;
import ch.unisg.backend.core.port.out.NonFunctionalRequirementPort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class NonFunctionalRequirementService implements NonFunctionalRequirementUseCase {

  private final NonFunctionalRequirementPort nonFunctionalRequirementPort;

  @Override
  public void create(NonFunctionalRequirementCommand command) {
    NonFunctionalRequirement nonFunctionalRequirement = NonFunctionalRequirement.create(command.id(), command.title(),
        command.isCyber());
    nonFunctionalRequirementPort.create(nonFunctionalRequirement);
  }

  @Override
  public NonFunctionalRequirement findById(NonFunctionalRequirementQuery query) {
    return nonFunctionalRequirementPort.findById(query.id());
  }
}

File: source/agent/source/application/non_functional_requirement_service.py
from source.application.domain.non_functional_requirement.enitity import NonFunctionalRequirement
from source.application.port.out.non_functional_requirement_port import NonFunctionalRequirementPort
from source.application.domain.non_functional_requirement.exception import NonFunctionalRequirementNotFound

class NonFunctionalRequirementService:
    def __init__(self, non_functional_requirement_port: NonFunctionalRequirementPort):
        self.non_functional_requirement_port = non_functional_requirement_port

    def get_non_functional_requirement(self, element_id: str) -> NonFunctionalRequirement:
        print("HO")
        raise NonFunctionalRequirementNotFound(f"{element_id} not found")
